plugins/node/opentelemetry-instrumentation-pino/src/pino.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  context,
+  diag,
+  getSpan,
+  isSpanContextValid,
+  Span,
+} from '@opentelemetry/api';
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  safeExecuteInTheMiddle,
+} from '@opentelemetry/instrumentation';
+import { Pino, PinoInstrumentationConfig } from './types';
+import { VERSION } from './version';
+import type * as pino from 'pino';
+
+const pinoVersions = ['>=5.14.0 <7'];
+
+export class PinoInstrumentation extends InstrumentationBase {
+  _isEnabled: boolean = false;
+
+  constructor(config: PinoInstrumentationConfig = {}) {
+    super('@opentelemetry/instrumentation-pino', VERSION, config);
+  }
+
+  protected init() {
+    return [
+      new InstrumentationNodeModuleDefinition<Pino>(
+        'pino',
+        pinoVersions,
+        pinoModule => {
+          const instrumentation = this;
+          const patchedPino = Object.assign((...args: unknown[]) => {
+            if (args.length == 0) {
+              return pinoModule({ mixin: instrumentation._getMixinFunction() });
+            }
+
+            if (args.length == 1) {
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              const optsOrStream = args[0] as any;
+              if (
+                typeof optsOrStream === 'string' ||
+                (optsOrStream !== undefined &&
+                  typeof optsOrStream['write'] === 'function')
+              ) {
+                args.splice(0, 0, {
+                  mixin: instrumentation._getMixinFunction(),
+                });
+                return pinoModule(...(args as Parameters<Pino>));
+              }
+            }
+
+            args[0] = instrumentation._combineOptions(
+              args[0] as pino.LoggerOptions
+            );
+
+            return pinoModule(...(args as Parameters<Pino>));
+          }, pinoModule);
+
+          return patchedPino;
+        },
+        () => {
+          this._isEnabled = false;
+        }
+      ),
+    ];
+  }
+
+  getConfig(): PinoInstrumentationConfig {
+    return this._config;
+  }
+
+  setConfig(config: PinoInstrumentationConfig) {
+    this._config = config;
+  }
+
+  enable() {
+    super.enable();
+    this._isEnabled = true;
+  }
+
+  disable() {
+    super.disable();
+    this._isEnabled = false;
```

Comment:
hmm we already have such functionality, can you add todo here to remove it once the base class will be changed. I created an issue for that -> https://github.com/open-telemetry/opentelemetry-js/issues/2131

Reply:
Added a TODO, found it kinda awkward as well :smile: